TombEngine/Sound/sound.cpp

```diff
@@ -192,77 +197,85 @@ bool SoundEffect(int effectID, Pose* position, SoundEnvironment condition, float
 	// Assign common sample flags.
 	DWORD sampleFlags = SOUND_SAMPLE_FLAGS;
 
-	// Effect's chance to play.
-	if ((sampleInfo->Randomness) && ((GetRandomControl() & UCHAR_MAX) > sampleInfo->Randomness))
+	// Test play chance.
+	if (sample.Randomness && ((GetRandomControl() & UCHAR_MAX) > sample.Randomness))
 		return false;
 
-	// Apply 3D attrib only to sound with position property
-	if (position)
+	// Apply 3D attribute only to sound with position property.
+	if (pose != nullptr)
 		sampleFlags |= BASS_SAMPLE_3D;
 
-	// Set & randomize volume (if needed)
-	float gain = (static_cast<float>(sampleInfo->Volume) / UCHAR_MAX) * std::clamp(gainMultiplier, SOUND_MIN_PARAM_MULTIPLIER, SOUND_MAX_PARAM_MULTIPLIER);
-	if ((sampleInfo->Flags & SOUND_FLAG_RND_GAIN))
-		gain -= (static_cast<float>(GetRandomControl()) / static_cast<float>(RAND_MAX))* SOUND_MAX_GAIN_CHANGE;
+	// Set and randomize volume (if needed).
+	float gain = ((float)sample.Volume / UCHAR_MAX) * std::clamp(gainMult, SOUND_MIN_PARAM_MULTIPLIER, SOUND_MAX_PARAM_MULTIPLIER);
+	if ((sample.Flags & SOUND_FLAG_RND_GAIN))
+		gain -= Random::GenerateFloat(0.0f, 1.0f) * SOUND_MAX_GAIN_CHANGE;
 
-	// Set and randomize pitch and additionally multiply by provided value (for vehicles etc)
-	float pitch = (1.0f + static_cast<float>(sampleInfo->Pitch) / 127.0f) * std::clamp(pitchMultiplier, SOUND_MIN_PARAM_MULTIPLIER, SOUND_MAX_PARAM_MULTIPLIER);
+	// Set and randomize pitch and additionally multiply by provided value (e.g. for vehicles).
+	float pitch = (1.0f + ((float)sample.Pitch / 127.0f)) * std::clamp(pitchMult, SOUND_MIN_PARAM_MULTIPLIER, SOUND_MAX_PARAM_MULTIPLIER);
 
 	// Randomize pitch (if needed)
-	if ((sampleInfo->Flags & SOUND_FLAG_RND_PITCH))
-		pitch += ((static_cast<float>(GetRandomControl()) / static_cast<float>(RAND_MAX)) - 0.5f) * SOUND_MAX_PITCH_CHANGE * 2.0f;
+	if ((sample.Flags & SOUND_FLAG_RND_PITCH))
+		pitch += Random::GenerateFloat(-0.5f, 0.5f) * (SOUND_MAX_PITCH_CHANGE * 2);
 
-	// Calculate sound radius and distance to sound
-	float radius = (float)(sampleInfo->Radius) * BLOCK(1);
-	float distance = Sound_DistanceToListener(position);
+	// Calculate sound radius and distance to sound.
+	float radius = BLOCK(sample.Radius);
+	float dist = Sound_DistanceToListener(pose);
```

Comment:
Usually you make variable names more verbose and descriptive, but here you decided to do the opposite and convert `distance` to `dist`. Why?

Reply:
I use the full long word in more formal contexts such as a class name, function name, or member variable, but for local variables I go for brevity. Same with pos/position, orient/orientation, rot/rotation, etc.

Reply:
But I still don't understand why this choice.

Reply:
Consistency.

Reply:
I understand that you try to maintain consistency with your own code in other places of the engine. However, it is not indicated in coding conventions, and this is the first time I am hearing this from you and seeing this.